### PR TITLE
Improve InMemoryProvider getRange function from O(N^2) to O(N)

### PR DESCRIPTION
**Pull request by bhargeymehta**
I noticed that in the getRange function we're doing a O(N^2) operation when constructing the results array. This time complexity arises because of doing a concat of 2 arrays instead of doing push operations on a single array.

### CODE REVIEW

File: src/InMemoryProvider.ts [hunk 1]
```diff
@@ -768,17 +768,16 @@ class InMemoryIndex extends DbIndexFTSFromRangeQueries {
             break;
           }
 
+          const pushToResult = (v: ItemType) => values.push(v);
```
Comment (thomastay) [inline src/InMemoryProvider.ts:771]: As `values` is no longer mutated, we should change it to be `const`

**Review by thomastay — CHANGES_REQUESTED** [review]
Other than the comment, lgtm, but please wait for other folks on my team to take a look as well before merging

**Comment by thomastay** [comment]
Can you also bump the patch version in the package.json

File: src/InMemoryProvider.ts (hunk 1 above)
Comment (bhargeymehta) [reply src/InMemoryProvider.ts:771]: changed to const

**Comment by bhargeymehta** [comment]
@microsoft-github-policy-service agree company="Microsoft"

**Comment by bhargeymehta** [comment]
> Can you also bump the patch version in the package.json

done

**Review by eliranek1 — APPROVED** [review]

**Review by thomastay — APPROVED** [review]

**Review by M1Les — APPROVED** [review]

**Pull request merged**